### ISSUE TITLE
Add target filtering for strict/unused dep tracking

### ISSUE THREAD

**Issue by liucijus** [bazelbuild/rules_scala#1240]
I did experiment with dep tracking filtering (#1222) and got great [results](https://github.com/bazelbuild/rules_scala/pull/1222#issuecomment-790428620). I think adding filtering would benefit other `rules_scala` users.

I'm thinking of adding the following attributes to the scala toolchain:
```starlark
"dependency_tracking_strict_deps_patterns": attr.string_list(
        doc = "List of target prefixes included for strict deps analysis",
        default = [""],
),
"dependency_tracking_unused_deps_patterns": attr.string_list(
        doc = "List of target prefixes included for unused deps analysis",
        default = [""],
),
```
Example configuration:
```starlark
dependency_tracking_strict_deps_patterns: [
    "@foo",      # track all targets which start with "@foo"
    "//",        # track every local target starting with "//"
    "-//:bar",   # but ignore "//:bar"
],
dependency_tracking_unused_deps_patterns: [
    "//foo/bar", # track only targets starting with "//foo/bar"
],
```

default configuration would be `[""]`, which matches every target.

Tracking decision would be made in starlark and is relatively cheap (check if target label `startswith` over the list items) and will cache well.

Calling for all types of input (naming, potential problems, design improvements, etc).

**Comment by liucijus** [comment]
Addresses #1046

**Comment by liucijus** [comment]
Solved with #1241
